tools/analyzer_plugin/lib/src/diagnostic/duplicate_prop_cascade.dart

```diff
@@ -27,11 +27,13 @@ class DuplicatePropCascadeDiagnostic extends ComponentUsageDiagnosticContributor
   @override
   computeErrorsForUsage(result, collector, usage) async {
     final propUsagesByName = groupBy<PropAssignment, String>(usage.cascadedProps, (prop) {
+      final enclosingElement = prop.name.staticElement?.enclosingElement?.name ?? '';
```

Comment:
Sorry, I just realized this could use a comment:

Also #nit on naming
```suggestion
      // Prefix with the class the prop was declared in so that things like `.dom.type` don't conflict with an unrelated `type` prop.
      final enclosingElementName = prop.name.staticElement?.enclosingElement?.name ?? '';
```